cmd/service.go

```diff
@@ -28,45 +32,67 @@ var serviceCmd = &cobra.Command{
 }
 
 var serviceCreateCmd = &cobra.Command{
-	Use:   "create <service_type> [service_name]",
+	Use:   "create <service_type> --plan <plan_name> [service_name]",
 	Short: "Create a new service",
-	Long: `Create a new service from service catalog to deploy on OpenShift.
+	Long: `Create a new service from service catalog using the plan defined and deploy it on OpenShift.
 
-If service name is not provided, service type value will be used.
+If service name is not provided, service type value will be used. The plan to be used must be passed along the service type
+using this convention <service_type>/<plan>. The parameters to configure the service are passed as a list of key=value pairs.
+The list of the parameters and their type is defined according to the plan selected.
 
 A full list of service types that can be deployed are available using: 'odo catalog list services'`,
-	Example: `  # Create new mysql-persistent service from service catalog.
-  odo service create mysql-persistent
+	Example: `  # Create new postgresql service from service catalog using dev plan and name my-postgresql-db.
+  odo service create dh-postgresql-apb my-postgresql-db --plan dev -p postgresql_user=luke -p postgresql_password=secret
 	`,
-	Args: cobra.RangeArgs(1, 2),
+	Args: cobra.RangeArgs(1, 1),
 	Run: func(cmd *cobra.Command, args []string) {
 		client := getOcClient()
 		applicationName, err := application.GetCurrentOrGetCreateSetDefault(client)
 		checkError(err, "")
 		projectName := project.GetCurrent(client)
+
+		// make sure the service type exists
 		serviceType := args[0]
-		exists, err := svc.SvcTypeExists(client, serviceType)
+		matchingService, err := svc.GetSvcByType(client, serviceType)
 		checkError(err, "unable to create service because Service Catalog is not enabled in your cluster")
-		if !exists {
+		if matchingService == nil {
 			fmt.Printf("Service %v doesn't exist\nRun 'odo service catalog' to see a list of supported services.\n", serviceType)
 			os.Exit(1)
 		}
+
+		// make sure that plan was supplied and exists
+		if len(plan) == 0 {
```

Comment:
Perhaps check this first to fail fast if needed?

Reply:
Sure, makes sense!

Reply:
On the other hand, by deferring until we know the allowed plans, we can supply a better error message. So I am not sure what the best experience is...

Reply:
Best experience is interactive mode! ;)

Reply:
Without a doubt!

Reply:
Agreed that resolving the possible plans is best for now.